tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/Net/GraphQLCommon.cs

```diff
@@ -147,37 +153,112 @@ private static string ConstructErrorMessage(IExecutionErrors executionErrors)
 
                 for (int i = 0; i < executionErrors.Count; i++)
                 {
+                    var eventAttributes = new List<KeyValuePair<string, object>>();
                     var executionError = executionErrors[i];
 
+                    if (executionErrors[i].Instance.TryDuckCast<IExecutionErrorExtensions>(out var executionErrorExtensions))
+                    {
+                        var extensions = executionErrorExtensions.Extensions;
+                        if (extensions != null)
+                        {
+                            var configuredExtensions = Tracer.Instance.Settings.GraphQLErrorExtensions;
+
+                            var keys = extensions.Keys.ToList();
```

Comment:
Also leaving a comment here, I recommend removing this `ToList()` call because it allocates and instead iterate over the `extensions` object itself (it already implements `IEnumerable<KeyValuePair<string, object>>`)